### DISCUSSION TITLE
[node.js, next.js] SSR and Supabase Storage

### DISCUSSION THREAD

**Discussion by almosnow** [supabase/supabase#27682] [Questions]
I am using @supabase/ssr in a next.js project (or for practical purposes, node.js). I like the idea of doing some stuff on the server.

I'm wondering if I could do the same with storage, not having the client spanning a supabase instance, but rather talk to a function my end and then I'm the one calling storage functions, pretty much proxying their data.

How may I accomplish this?

**Comment by silentworks** [comment]
It's not very clear what you are asking here. But if you are using `@supabase/ssr` package you can access all services on the server except for realtime.

**Reply by almosnow** [reply]
You're right, I have to be clearer,

So, I can call the storage functions from the backend, but I would like to avoid storing the data and then have to send it to supabase.

Instead of:

1. client (browser) calls myRESTAPI.upload(file) to upload a file.
2. I download/store the file in the backend and then call supabase.storage.[...].upload(file).

I would want to set up something like:

1. client (browser) calls myRESTAPI.upload(file) to upload a file.
2. I proxy that call/data to a supabase.storage.[...].upload(file) call w/ the appropriate credentials. This is transparent to the client and I just move bytes around.

And, yes, I'm using @supabase/ssr.

I'm sure it could be done. I ask for advice here as there's probably a two-liner that accomplishes that already, 🙂.
